presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -122,6 +125,15 @@ public long getWrittenBytes()
             return ((ExtendedRecordWriter) recordWriter).getWrittenBytes();
         }
 
+        if (committed) {
+            try {
+                return path.getFileSystem(conf).getFileStatus(path).getLen();
```

Comment:
`ExtendedRecordWriter` reports `writtenBytes` on the go. I would leave it as is.

Reply:
Yes, i noticed. Why not improve default record writer to report written bytes on the go as well? Still, according to the commit message that introduced the `ExtendedRecordWriter`, the main difference lied in `path.getFileSystem(conf).getFileStatus(path).getLen();` -- which is no longer a difference.

Reply:
> Why not improve default record writer to report written bytes on the go as well? 

DefaultWriter uses writers from Hive. Not all of them report the writen bytes statistic.

Reply:
Oh, i thought `ExtendedRecordWriter` is a Hive class. Yeah, let me check that.

Reply:
So, basically we use this for `RCFile.Writer` because specifically `RCFile.Writer` provides `#getLength()` method that can be used to get bytes written. We do `Object writer = Class.forName(outputFormatName).getConstructor().newInstance();` when getting all the other writers. So we would have to do a big switch instead and apply `ExtendedFileWriter` whenever it is possible. Although we got not that many supported formats, it still feels to be tedious.